### PR TITLE
Shutdown toolset compilers

### PR DESCRIPTION
**Pull request by jjonescz**
Part of https://github.com/dotnet/roslyn/issues/72672.

### CODE REVIEW

**Review by jaredpar — APPROVED** [review]

File: src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs [hunk 1]
```diff
@@ -3,12 +3,18 @@
...
 
 namespace Microsoft.DotNet.BuildServer
 {
     internal class VBCSCompilerServer : IBuildServer
     {
+        private static readonly string s_toolsetPackageName = "microsoft.net.sdk.compilers.toolset";
+        private static readonly string s_vbcsCompilerExeFileName = "VBCSCompiler.exe";
+        private static readonly string s_shutdownArg = "-shutdown";
+
         internal static readonly string VBCSCompilerPath = Path.Combine(
```
[the remark this answers is not in the document]
Comment (jjonescz) [reply src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs:18]: It's used in tests currently, I can add a comment.

[the remark this answers is not in the document]
Comment (jaredpar) [reply src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs:18]: No need for comment. Was just curious.

File: src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs
Comment (jjonescz) [inline src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs:1]: (note that this PR targets release/9.0.1xx, but I'm unsure that's correct)

**Comment by jjonescz** [comment]
@rainersigwald for a review, thanks

**Comment by jjonescz** [comment]
@rainersigwald for a second review, thanks

**Review by rainersigwald — APPROVED** [review]

File: src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs
Comment (marcpopMSFT) [reply src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs:1]: Where did you want to ship this? 9.0.1xx is still open for checkins unless they are breaking changes or high risk.

Comment (jjonescz) [reply src/Cli/dotnet/BuildServer/VBCSCompilerServer.cs:1]: Ideally .NET 9 so it's shipped with the other torn sdk changes, I wasn't sure it meets the bar, but it shouldn't be high risk.

**Pull request merged**